### PR TITLE
Optimize 'receive_json' for the happy path

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/impl/shard.py [hunk 1]
```diff
@@ -177,53 +177,60 @@ async def send_json(
         await self.send_str(pl, compress)
 
     async def _receive_and_check(self, timeout: typing.Optional[float], /) -> str:
-        buff = bytearray()
+        message = await self.receive(timeout)
 
-        while True:
-            message = await self.receive(timeout)
+        if message.type == aiohttp.WSMsgType.TEXT:
+            return message.data  # type: ignore
```
Comment (FasterSpeeding) [inline hikari/impl/shard.py:183]: why not assert the type?

**Comment by FasterSpeeding** [comment]
LGMT though

File: hikari/impl/shard.py (hunk 1 above)
Comment (davfsa) [reply hikari/impl/shard.py:183]: This was already like this, I only moved code around. Can be changed tho

Comment (FasterSpeeding) [reply hikari/impl/shard.py:183]: Yeah I'd appreciate changing it even if its not really the purpose of this pr 👍

Comment (davfsa) [reply hikari/impl/shard.py]: Added

**Review by FasterSpeeding — DISMISSED** [review]

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**